plugins/CorePluginsAdmin/PluginInstaller.php

```diff
@@ -297,12 +298,20 @@ private function fixPluginFolderIfNeeded($tmpPluginFolder)
 
     private function copyPluginToDestination($tmpPluginFolder)
     {
-        $pluginsDir = Manager::getPluginsDirectory();
-        $pluginTargetPath = $pluginsDir . $this->pluginName;
-
-        $this->removeFolderIfExists($pluginTargetPath);
-
-        Filesystem::copyRecursive($tmpPluginFolder, $pluginsDir);
+        if (!empty(Config::getInstance()->General['copy_to_plugins_dirs'])) {
+            $pluginsDirs = Manager::getPluginsDirectories();
+            // Copy plugin to all plugins directories
+            foreach ($pluginsDirs as $dir) {
```

Comment:
this copies it into all plugin directories, I reckon matomo may have trouble with this if there are multiple directories defined. It should copy the plugin only into one plugin directory. We have uses cases where we define multiple plugin directories (one directory for each plugin in fact)

Reply:
i changed to copy only to directory defined in `MATOMO_PLUGIN_COPY_DIR` if it's within `getPluginsDirectories ` array